American Whitewater/Helpers/DefaultsManager.swift

```diff
@@ -79,8 +79,7 @@ class DefaultsManager {
     public var filters: Filters {
         get {
             .init(
-                showDistanceFilter: showDistanceFilter,
-                showRegionFilter: showRegionFilter,
+                filterType: filterType,
```

Comment:
OK, I have 1 substantive comment on these changes (which overall I LOVE)

Swapping these over without providing a migration path is going to reset users' filter distance/region filter choice when it ships. Is that an OK thing to do? It might be confusing, but on the other hand, it's pretty straightforward to switch it back and they aren't losing any data (beyond a filter preference).

If we wanted to keep it consistent, filterType's getter in DefaultsManager could check `showDistanceFilter` and use that to set an initial value. Downside of that approach is you get a less tidy migration, and the app has this bit of cruft checking an otherwise deleted user defaults key.

Reply:
That's a good point. These sorts of upgrade paths are fraught with pitfalls though, and I don't think we've got the resources to tackle them. If I'm understanding correctly, the only user effect will be, if they have a distance filter, it will be reset to a region filter. That might get into an annoying state where they then need to go in to select a region or get that "all regions" warning, but I think that's better and more consistent than having to leave the old defaults in the code.